packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb.tmpl

```diff
@@ -42,30 +42,12 @@ end
 #   target 'MyApp' do
 #     install_flutter_engine_pod
 #   end
-def install_flutter_engine_pod
-  current_directory = File.expand_path('..', __FILE__)
-  engine_dir = File.expand_path('engine', current_directory)
-  framework_name = 'Flutter.xcframework'
-  copied_engine = File.expand_path(framework_name, engine_dir)
-  if !File.exist?(copied_engine)
-    # Copy the debug engine to have something to link against if the xcode backend script has not run yet.
-    # CocoaPods will not embed the framework on pod install (before any build phases can generate) if the dylib does not exist.
-    release_framework_dir = File.join(flutter_root, 'bin', 'cache', 'artifacts', 'engine', 'ios-release')
-    unless Dir.exist?(release_framework_dir)
-      # iOS artifacts have not been downloaded.
-      raise "#{release_framework_dir} must exist. Make sure \"flutter precache --ios\" has been run at least once"
-    end
-    FileUtils.cp_r(File.join(release_framework_dir, framework_name), engine_dir)
-  end
-
-  # Keep pod path relative so it can be checked into Podfile.lock.
-  # Process will be run from project directory.
-  engine_pathname = Pathname.new engine_dir
-  # defined_in_file is a Pathname to the Podfile set by CocoaPods.
-  project_directory_pathname = defined_in_file.dirname
-  relative = engine_pathname.relative_path_from project_directory_pathname
+def install_flutter_engine_pod(flutter_application_path = nil)
+  flutter_application_path ||= File.join('..', '..')
+  ios_application_path = File.join(flutter_application_path, '.ios')
 
-  pod 'Flutter', :path => relative.to_s, :inhibit_warnings => true
+  # flutter_install_ios_engine_pod is in Flutter root podhelper.rb
+  flutter_install_ios_engine_pod(ios_application_path)
```

Comment:
Instead of duplicating logic, call the ruby script in the tool 
https://github.com/flutter/flutter/blob/aa5d7b6972ee5be596c15b83d39f7d3dde6ad939/packages/flutter_tools/bin/podhelper.rb#L159